gtsam/basis/basis.i

```diff
@@ -122,3 +126,32 @@ class ManifoldEvaluationFactor : gtsam::NoiseModelFactor {
                            const gtsam::noiseModel::Base* model, const size_t N,
                            double x, double a, double b);
 };
+
+// TODO(gerry): Add `DerivativeFactor`, `VectorDerivativeFactor`, and
+// `ComponentDerivativeFactor`
+
+#include <gtsam/basis/FitBasis.h>
+// We'll allow transparent binding of python dict to Sequence in this
+// compilation unit using pybind11/stl.h.
+// Another alternative would be making Sequence opaque in
```

Comment:
Pick one option and implement that?

Reply:
Thanks, I was thinking someone might question why I went this route including `pybind11/stl.h` so I wanted to give a reason why, but in the future I'll be less verbose.

Reply:
Verbosity is not the issue, the more verbose the better actually. I am more of the opinion that we should pick one way and stick to it, and the alternatives can be discussed beforehand. :slightly_smiling_face: 

FYI I am adding the stl.h include to the `gtsam.tpl` file directly since includes don't add much overhead.

Reply:
Actually no, we don't include `stl.h` since it interferes with other containers where we need reference passing. I'm just adding `Sequence` to specializations.

Reply:
I got build errors when I tried that, hence only including it in the basis module - but would be helpful if you did succeed :)

I assume related to other folders using opaque types

Reply:
oops commented at the same time - ya

My reasoning for not adding Sequence to specializations is described in this comment: it will muddy the syntax to do

```python
fit = gtsam.FitBasis(3, gtsam.Sequence({data}), noiseModel)
```
as compared to 
```python
fit = gtsam.FitBasis(3, data, noiseModel)
```

And the advantages of the specialization stuff, namely correct reference passing, isn't as important because this is never going to be a mutable variable.

Maybe I should have been even more verbose in my comment to make this clearer 😂